src/storage/StateSnapshotStorage.ts

```diff
@@ -0,0 +1,81 @@
+import { ethers } from "ethers";
+import { Codec, Type } from "@/utils";
+import { StateSnapshotStruct } from "@typechain-types/contracts/V1/DataTypes";
+import { IStateSnapshotStorageModule } from "./interfaces/IStateSnapshotStorage";
+import { StateSnapshotHash } from "@/types/storage";
+
+export class StateSnapshotStorageModule implements IStateSnapshotStorageModule {
+    //map [stateSnapshotHash]=> struct
+    // we make the key a string to avoid double mapping
+    private stateSnapshotStructsMap: Map<
+        StateSnapshotHash,
+        StateSnapshotStruct
+    >;
+    //could be replaced with a key, but we would need some computation
+    private cachedOnChainStateSnapshot:
```

Comment:
Similar analogy to 'latestBlockConfirmation' - doesn't need to be here. I agree this information is useful, the snapshot is persisted regardless and some other component can handle the business logic/caching what's the onChainSnapshot - maybe StorageManager can keep the hash and when needed retrieve the specific snapshot from here